independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverInfo.java

```diff
@@ -99,6 +101,17 @@ public boolean isAsync() {
         return isAsync;
     }
 
+    public Reception getReception() {
+        AnnotationInstance observesAnnotation = isAsync
+                ? declaringBean.getDeployment().getAnnotation(observerMethod, DotNames.OBSERVES_ASYNC)
```

Comment:
I am not sure this makes much sense for async because as per spec, request context is active during notifications of async observers and that leaves us with app/singleton scoped which are always active. Hence the behaviour will be the same as if you always had `Reception.ALWAYS` if I am not mistaken?
It might be worth something if more scopes are in play but I am not sure we support custom ones at this point?

Reply:
I think you're wrong. `Reception.IF_EXISTS` means that we should only deliver the notification if a contextual instance already exists in the current context.

Reply:
And it seems I have a bug there ;-)

Reply:
Then I might have misunderstood your changes in `ObserverGenerator`. I was under the impression that you check for context being active, not an existing instance?

Reply:
As I said above - there is a bug.

Reply:
Missed that comment before, sorry :)